llvm/test/Transforms/LoopIdiom/cheri-preserve-tags-store.ll

```diff
@@ -0,0 +1,99 @@
+; PURECAP:       for.end:
+; PURECAP-NEXT:    ret void
+;
+entry:
+  br label %for.body.preheader
+
+for.body.preheader:                                   ; preds = %0
+  br label %for.body
+
+for.body:                                         ; preds = %entry, %for.body
+  %i.08 = phi i64 [ 0, %for.body.preheader ], [ %inc, %for.body ]
+  %arrayidx = getelementptr inbounds [25 x i32], ptr addrspace("G") @nocap, i64 0, i64 %i.08
+  %0 = load i32, ptr addrspace("A") %arrayidx, align 4
+  %arrayidx2 = getelementptr inbounds %struct.state, ptr addrspace("A") %state, i64 0, i32 0, i64 %i.08
+  store i32 %0, ptr addrspace("A") %arrayidx2, align 4
+  %inc = add nuw nsw i64 %i.08, 1
+  %exitcond = icmp ne i64 %inc, 25
+  br i1 %exitcond, label %for.body, label %for.end
+
+for.end:                                          ; preds = %for.body
+  ret void
+}
+
+define void @get_cap_state(ptr addrspace("A") nocapture noalias %state) local_unnamed_addr addrspace("P") #0 {
+; HYBRID-LABEL: @get_cap_state(
+; HYBRID-NEXT:  entry:
+; HYBRID-NEXT:    call void @llvm.memcpy.p0.p0.i64(ptr align 16 [[STATE:%.*]], ptr align 16 @cap, i64 400, i1 false) #[[ATTR2:[0-9]+]]
+; HYBRID-NEXT:    ret void
+;
+; PURECAP-LABEL: @get_cap_state(
+; PURECAP-NEXT:  entry:
+; PURECAP-NEXT:    br label [[FOR_BODY:%.*]]
+; PURECAP:       for.body:
+; PURECAP-NEXT:    [[I_08:%.*]] = phi i64 [ 0, [[ENTRY:%.*]] ], [ [[INC:%.*]], [[FOR_BODY]] ]
+; PURECAP-NEXT:    [[ARRAYIDX:%.*]] = getelementptr inbounds [25 x ptr addrspace(200)], ptr addrspace(200) @cap, i64 0, i64 [[I_08]]
+; PURECAP-NEXT:    [[TMP0:%.*]] = load ptr addrspace(200), ptr addrspace(200) [[ARRAYIDX]], align 16
+; PURECAP-NEXT:    [[ARRAYIDX2:%.*]] = getelementptr inbounds [[STRUCT_CAPSTATE:%.*]], ptr addrspace(200) [[STATE:%.*]], i64 0, i32 0, i64 [[I_08]]
+; PURECAP-NEXT:    store ptr addrspace(200) [[TMP0]], ptr addrspace(200) [[ARRAYIDX2]], align 16
+; PURECAP-NEXT:    [[INC]] = add nuw nsw i64 [[I_08]], 1
+; PURECAP-NEXT:    [[EXITCOND:%.*]] = icmp ne i64 [[INC]], 25
+; PURECAP-NEXT:    br i1 [[EXITCOND]], label [[FOR_BODY]], label [[FOR_END:%.*]]
+; PURECAP:       for.end:
+; PURECAP-NEXT:    ret void
+;
+entry:
+  br label %for.body
+
+for.body:                                         ; preds = %entry, %for.body
+  %i.08 = phi i64 [ 0, %entry ], [ %inc, %for.body ]
+  %arrayidx = getelementptr inbounds [25 x ptr addrspace(200)], ptr addrspace("G") @cap, i64 0, i64 %i.08
+  %0 = load ptr addrspace(200), ptr addrspace("G") %arrayidx, align 16
+  %arrayidx2 = getelementptr inbounds %struct.capstate, ptr addrspace("A") %state, i64 0, i32 0, i64 %i.08
+  store ptr addrspace(200) %0, ptr addrspace("A") %arrayidx2, align 16
+  %inc = add nuw nsw i64 %i.08, 1
+  %exitcond = icmp ne i64 %inc, 25
+  br i1 %exitcond, label %for.body, label %for.end
+
+for.end:                                          ; preds = %for.body
+  ret void
+}
+
+; UTC_ARGS: --disable
+; HYBRID: attributes #[[ATTR1]] = { no_preserve_cheri_tags }
```

Comment:
These test checks are quite awkward and not great for understanding. I wonder if it would make more sense to have this attribute apply to the destination argument of the call?